hw4/02-sessions.js

```diff
@@ -1,15 +1,38 @@
-const express = require('express');
-const session = require('express-session');
+const express = require("express");
+const session = require("express-session");
 const app = express();
 const port = process.env.PORT || 5001;
 
 // Add your code here
+app.use(
+  session({
+    store: new session.MemoryStore(),
+    secret: "abcd",
+    resave: false,
+    saveUninitialized: true
+  })
+);
 
-// Use the express-session module
-app.use(/** ... */);
+app.get("/favicon.ico", (req, res) => res.status(204));
 
-app.get('/', (req, res) => {
-  // Add your code here
+app.use((req, res, next) => {
+  res.setHeader("content-type", "text/html");
+  res.write(`<p>Currently on route: ${req.url}</p>`);
+  if (req.session.previousUrls) {
+    req.session.previousUrls.push(`${req.url}`);
```

Comment:
Here when a user visits a route it's also showing up in the "Previously Visited" section because you're pushing it to the previousUrls array before writing it to the response. Based on the screenshots you should not push it to the previousUrls array until after you write it